### PR TITLE
TRSV2-175: forgotten password

### PR DESCRIPTION
**Pull request by KnowYourLines**

### CODE REVIEW

File: trade_remedies_public/templates/password/password_reset_requested.html [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
-{% endblock %}
+{% block fieldset_heading %}
+Password reset email sent
+{% endblock fieldset_heading %}
 
 {% block form_content %}
-    <p>
-      Please check your email and click the link provided to reset your password.
-    </p>
+<br/><br/>
```
Comment (chris-pettinga) [inline trade_remedies_public/templates/password/password_reset_requested.html:13]: Does the prototype have these breaks?

File: trade_remedies_public/templates/password/password_reset_requested.html [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
+<p>We've sent an email to the email address you provided. Use the link in the email to reset your password.</p>
```
Comment (chris-pettinga) [inline trade_remedies_public/templates/password/password_reset_requested.html:14]: Do you need to use the govuk-body class?

File: trade_remedies_public/templates/password/reset_password_request.html [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
+Forgotten your password?
+{% endblock fieldset_heading %}
 
 {% block form_content %}
-    <form action="{% url 'forgot_password' %}" method="post">
-        {% csrf_token %}
-        {% text_element id='email' label='Email address' errors=errors %}
-        <button type="submit" class="button">Request password reset</button>
-    </form>
+<br/><br/>
```
Comment (chris-pettinga) [inline trade_remedies_public/templates/password/reset_password_request.html:13]: Same comment as above

**Review by chris-pettinga — CHANGES_REQUESTED** [review]
Looks good, just some comments on HTML/JS reusability

File: trade_remedies_public/config/middleware.py [hunk 4]
```diff
@@ -16,6 +16,15 @@
     reverse("logout"),
 )
 
+NON_BACK_URLS = [reverse("landing"), reverse("reset_password_success")]
+
+
+def get_evaluated_non_back_urls(kwargs):
+    urls = NON_BACK_URLS
+    if kwargs.get("user_pk") and kwargs.get("token"):
```
[the remark this answers is not in the document]
Comment (KnowYourLines) [reply trade_remedies_public/config/middleware.py:24]: We don't want a back button on the reset_password page for the happy path, but we can't get that URL if those two arguments are not present. An error will be thrown by `reverse`. This ensures we only try to reverse the URL if the arguments are present.

File: trade_remedies_public/templates/password/password_reset_requested.html (hunk 1 above)
Comment (KnowYourLines) [reply trade_remedies_public/templates/password/password_reset_requested.html:13]: `base_with_form.html` is a bit of premature optimization in this case as it isn't used in other parts of the prototype.

Comment (KnowYourLines) [reply trade_remedies_public/templates/password/password_reset_requested.html:13]:
```            <fieldset class="govuk-fieldset">
                <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
                    <h1 class="govuk-fieldset__heading">
                        {% block fieldset_heading %}{% endblock fieldset_heading %}
                    </h1>
                </legend>
            </fieldset>
```
This isn't common across other pages.

File: trade_remedies_public/templates/password/reset_password_request.html (hunk 3 above)
Comment (KnowYourLines) [reply trade_remedies_public/templates/password/reset_password_request.html:13]: base_with_form.html does not fit this template so might be a bit of a premature optimization

**Review by chris-pettinga — APPROVED** [review]

**Pull request merged**